tensorflow/lite/kernels/internal/tensor.h

```diff
@@ -72,7 +73,13 @@ class VectorOfTensors {
   // example:
   //   float* const* f = v.data();
   //   f[0][1] is the second element of the first tensor.
-  T* const* data() const { return all_data_.data(); }
+  // NOTE: This function should be called only in Eval/Invoke context
+  T* const* data() {
+    for (auto it = all_tensors_.begin(); it != all_tensors_.end(); ++it) {
+      all_data_.push_back(GetTensorData<T>(*it));
```

Comment:
Could you explain a bit about the benefit of refactoring this function?

In your current implementation, I guess if data() is called multiple times, the all_data_ will be appended with same amount of elements again and again? Is that expected behavior?

Reply:
> In your current implementation, I guess if data() is called multiple times, the all_data_ will be appended with same amount of elements again and again? Is that expected behavior?

@haozha111 : Sorry i missed that one, Thanks for pointing out, will resolve it.

Reply:
> Could you explain a bit about the benefit of refactoring this function?

@haozha111 : The main benefit is, it avoids heap allocation in Eval / Invoke time. Earlier it was allocating every time invoked, now it will be allocated only once, and will be reused every  Eval / Invoke time.

Reply:
Right. Sorry my question is not clear. I mean why do you need change 'VectorOfTensors' class? Does adding a new class member 'all_tensors_' have any benefit compared with before?

Reply:
@haozha111 : Yes, earlier 'VectorOfTensors' was reading data directly from Tensors, which will not be available in case of Prepare, so it was a limitation to create instance of 'VectorOfTensors' only in Eval. But now with my change, 'VectorOfTensors'  can be instantiated in Prepare as it stores all the Input Tensors, rather than their data. So when 'VectorOfTensors'->data() is called in Eval, it will fetch all the fresh data loaded into Tensors.

Reply:
> > In your current implementation, I guess if data() is called multiple times, the all_data_ will be appended with same amount of elements again and again? Is that expected behavior?
> 
> @haozha111 : Sorry i missed that one, Thanks for pointing out, will resolve it.

This issue is addressed now, please check, Thanks!

Reply:
Let's hold on this change a bit. I think this change will make the call a little expensive. The goal of 'VectorOfTensors' should be high efficient. 

Originally VectorOfTensors only handle static tensors (the data won't change), so it's simpler and more efficient. Now as you are supporting this in Prepare stage, where the data of tensor might not be ready. I think we might think a little bit about the trade off here, whether we want to keep the current semantics of the class(efficient, static view of tensors data), or changing its behavior to support dynamic view of tensors data.

Reply:
> Let's hold on this change a bit. I think this change will make the call a little expensive. The goal of 'VectorOfTensors' should be high efficient.
> 
> Originally VectorOfTensors only handle static tensors (the data won't change), so it's simpler and more efficient. Now as you are supporting this in Prepare stage, where the data of tensor might not be ready. I think we might think a little bit about the trade off here, whether we want to keep the current semantics of the class(efficient, static view of tensors data), or changing its behavior to support dynamic view of tensors data.

@haozha111 : Sure i will wait for your go ahead. BUt as concern raised by you, just wanted to share my view, if it can help. Actually performance wise, there is no change at all, if you still keep the instantiation in Eval(). But if instantiation kept in Prepare() and data() fetch in Eval(), it actually helps reducing memory allocation every time, performance wise it will still be same or minor improved. Only case where i can see it can go dip is when the data() fetch called multiple times in same sequence where Tensors are not changed at all. As every time it has to refresh and load data. 

Hope my above details helps a bit for your conclusion, Thanks!
Thanks again for your efforts in reviewing my changes!